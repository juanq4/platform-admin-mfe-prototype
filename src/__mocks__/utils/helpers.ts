import { isEmpty } from "@q4/nimbus-ui";

export function mockWindowMethods(): Window {
  return Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
}

export function mockClipboardWrite(): void {
  const orginalClipboard = { ...navigator.clipboard };

  beforeAll(() => {
    if (!isEmpty(navigator.clipboard)) return;

    Object.defineProperty(navigator, "clipboard", {
      value: {
        writeText: () => Promise.resolve(),
      },
    });
  });

  afterAll(() => {
    Object.assign(navigator.clipboard, {
      ...orginalClipboard,
    });
  });
}
