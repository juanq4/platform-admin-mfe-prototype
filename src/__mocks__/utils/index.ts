export * from "./helpers";
export * from "./permissions";
export * from "./wrappers";
