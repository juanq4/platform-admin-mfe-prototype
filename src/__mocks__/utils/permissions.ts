import { Permission } from "@q4/platform-definitions";

export const MockQ4AdminPermissions = [
  Permission.ReadUsers,
  Permission.ManageUsers,
  Permission.ReadOrgs,
  Permission.ManageOrgs,
];

export const MockQ4SupportPermissions = [Permission.ReadUsers, Permission.ReadOrgs, Permission.ManageOrgs];

export const MockCorporateAdminPermissions = [Permission.ReadUsers, Permission.ManageUsers];

export const MockCorporateSupportPermissions: Permission[] = [];
