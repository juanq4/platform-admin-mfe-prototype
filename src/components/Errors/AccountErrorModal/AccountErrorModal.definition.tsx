export const accountErrorTitle = "There is a problem with your account";
