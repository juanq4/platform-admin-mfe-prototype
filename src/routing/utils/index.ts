export * from "./api";
export * from "./s3checks";
export * from "./versionManager";
