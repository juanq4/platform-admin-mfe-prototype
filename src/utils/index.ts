export * from "./account";
export * from "./api";
export * from "./entitlement";
export * from "./error";
export * from "./feature";
export * from "./logout";
export * from "./organization";
export * from "./permission";
export * from "./role";
export * from "./route";
export * from "./subscription";
export * from "./user";
export * from "./environment";
