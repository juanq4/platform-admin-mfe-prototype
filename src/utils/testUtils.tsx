/* eslint-disable import/export */
import { MockedProvider as ApolloMockedProvider } from "@apollo/client/testing";
import { ThemeProvider } from "@emotion/react";
import { StyleGuide, ConfigProvider as NimbusConfig } from "@q4/nimbus-ui";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import theme from "../styles/theme";

// https://testing-library.com/docs/react-testing-library/setup/
const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
  styleGuide = StyleGuide.V2,
): RenderResult =>
  render(ui, {
    // Common middleware to all tests
    wrapper: ({ children }) => (
      <NimbusConfig styleGuide={styleGuide}>
        <ApolloMockedProvider>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </ApolloMockedProvider>
      </NimbusConfig>
    ),
    ...options,
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
