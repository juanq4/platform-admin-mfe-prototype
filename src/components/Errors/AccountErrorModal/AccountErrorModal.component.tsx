import { useAuth0 } from "@auth0/auth0-react";
import { Anchor, AnchorTheme, ButtonTheme, Modal, Text } from "@q4/nimbus-ui";
import { Email } from "../../../configurations";
import { logoutAndUpdateLocalStorage } from "../../../utils";
import { accountErrorTitle } from "./AccountErrorModal.definition";

export const AccountErrorModal = (): JSX.Element => {
  const { logout } = useAuth0();

  return (
    <Modal
      visible
      title={accountErrorTitle}
      footerActions={[
        {
          label: "Log out",
          theme: ButtonTheme.Primary,
          onClick: () => logoutAndUpdateLocalStorage(logout),
        },
      ]}
      includeExit={false}
      includeCancel={false}
    >
      <Text>
        Please contact{" "}
        <Anchor theme={AnchorTheme.Primary} url={`mailto:${Email.Support}`}>
          {Email.Support}
        </Anchor>{" "}
        for assistance.
      </Text>
    </Modal>
  );
};
