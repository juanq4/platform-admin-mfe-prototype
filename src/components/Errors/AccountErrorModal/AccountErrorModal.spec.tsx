import { render, screen } from "../../../utils/testUtils";
import { AccountErrorModal } from "./AccountErrorModal.component";

describe("AccountErrorModal", () => {
  it("7386735: renders component", () => {
    const { container } = render(<AccountErrorModal />);
    expect(container).toMatchSnapshot();
  });

  describe("Error Pages", () => {
    test("4326940: [Given] authenticated user sees account issues error modal [Then] expect 'Logout' button to be present on the error modal", () => {
      render(<AccountErrorModal />);
      expect(screen.getByText("Log out")).toBeVisible();
    });
  });
});
