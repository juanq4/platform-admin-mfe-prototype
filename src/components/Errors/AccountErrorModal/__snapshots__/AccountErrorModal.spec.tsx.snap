// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountErrorModal 7386735: renders component 1`] = `
<div
  aria-hidden="true"
  data-focus-on-hidden="true"
/>
`;
